### PR TITLE
adds conditions for selected and disabled for rerendering

### PR DESCRIPTION
**Pull request by gmzjuliana**
## Description
Adds disabled and selected as fields for comparison so that if they change, the items change accordingly

## Motivation and Context
I noticed while working in Analyze, that if the `selected` and `disabled` fields in items change, the select options aren't rerendering properly

## Screenshots (if appropriate):
Without the rerendering: https://share.buffer.com/8LuZYm6r
with the change: https://share.buffer.com/Z4uzGegY

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style and guidelines of this project. 
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have updated the **CHANGELOG** document.
- [ ] I have added tests to cover my changes.
- [x] I have performed a self-review of my own code
- [x] I have tested my changes/additions in the latest Chrome, Firefox, and Safari.
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] All new and existing tests passed.
- [x] I have performed the accessibility audit of my UI changes according to the accessibility doc. 
- [ ] [Buffer Engineers] Someone from the Design team reviewed and approved my changes
- [x] [Buffer Engineers] I have notified the BDS team of my changes in the #proj-design-system Slack channel

### CODE REVIEW

File: src/components/Select/Select.jsx [hunk 1]
```diff
@@ -29,7 +29,9 @@ export default class Select extends React.Component {
         el &&
         itemsB[ix] &&
         el.id === itemsB[ix].id &&
-        el.title === itemsB[ix].title
+        el.title === itemsB[ix].title &&
+        el.selected === itemsB[ix].selected &&
+        el.disabled === itemsB[ix].disabled
```
Comment (gmzjuliana) [inline src/components/Select/Select.jsx]: adding these like this doesn't feel great to me but I wonder if it's really worth comparing both objects key by key... I've love some advice here, if possible :)

Comment (msanroman) [reply src/components/Select/Select.jsx]: This feels good to me, to be honest :) This way we can be very specific on what we use to consider two select items equal or not ^_^

If we wanted to make it shorter and more generic, we could go with something like:

```javascript
Object.keys(el).every(attribute => el[attribute] === itemsB[ix][attribute])
```

**Comment by oscoda** [comment]
Thanks @gmzjuliana - also for the excellent video examples for context! 🙌🌮

File: src/components/Select/Select.jsx (hunk 1 above)
Comment (gmzjuliana) [reply src/components/Select/Select.jsx]: Thanks for the advice @msanroman 😁 I'll leave it explicit for now and we can keep it in mind for the future if we see a need to make it more generic

**Pull request merged**